as/cNFT/assembly/mint.ts

```diff
@@ -1,30 +1,32 @@
 import { context, logging, u128 } from 'near-sdk-as'
 import { assert_deposit_attached } from './utils/asserts'
-import { TokenMetadata } from '../../NFT/assembly/metadata'
-import { Token } from './models/token'
+import { Token } from './models/persistent_tokens'
 import { persistent_tokens } from './models/persistent_tokens'
+import { persistent_tokens_metadata, TokenMetadata } from './models/persistent_tokens_metadata'
 
 @nearBindgen
 export function mint(tokenMetadata: TokenMetadata): Token {
-    /** TODO Assert deposit attached based on custom amount from init */
+    /**@todo Assert deposit attached based on custom amount from init */
     //     assert_deposit_attached(u128.from(0))
 
-    /** TODO Assert uniqueId is actually unique */
-
     let token = new Token()
 
-    token.media = tokenMetadata.media
-    token.extra = tokenMetadata.extra
-    /** TODO not always */
+    /**@todo Not always sender is creator i guess, would rather pass it to function */
     token.creator = context.sender
```

Comment:
Here we are filling only the `creator` in token. But token has several properties like `id`, `owner_id` and `prev_owner`. So we are returning token without those properties. Is it okay ?